### ISSUE TITLE
NullPointerException by click on "load extended result" button

### ISSUE THREAD

**Issue by FelixNeutatz** [HPI-Information-Systems/Metanome#378]
When I click the "load extended result" button I get the following exception:

```
java.lang.NullPointerException
	at de.metanome.backend.result_postprocessing.helper.ColumnInformation.computeAverageLength(ColumnInformation.java:307)
	at de.metanome.backend.result_postprocessing.helper.ColumnInformation.computeDataDependentStatistics(ColumnInformation.java:117)
	at de.metanome.backend.result_postprocessing.helper.ColumnInformation.<init>(ColumnInformation.java:98)
	at de.metanome.backend.result_postprocessing.helper.TableInformation.<init>(TableInformation.java:83)
	at de.metanome.backend.result_postprocessing.result_analyzer.ResultAnalyzer.<init>(ResultAnalyzer.java:49)
	at de.metanome.backend.result_postprocessing.result_analyzer.FunctionalDependencyResultAnalyzer.<init>(FunctionalDependencyResultAnalyzer.java:41)
	at de.metanome.backend.result_postprocessing.ResultPostProcessor.analyzeAndStoreResults(ResultPostProcessor.java:261)
	at de.metanome.backend.result_postprocessing.ResultPostProcessor.extractAndStoreResults(ResultPostProcessor.java:177)
	at de.metanome.backend.result_postprocessing.ResultPostProcessor.extractAndStoreResultsDataDependent(ResultPostProcessor.java:77)
	at de.metanome.backend.resources.ResultStoreResource.loadExecution(ResultStoreResource.java:100)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:151)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:172)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$VoidOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:136)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:104)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:384)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:342)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:101)
	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:303)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:316)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:286)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1073)
	at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:377)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:381)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:344)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:221)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:230)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
	at org.apache.catalina.filters.CorsFilter.handleSimpleCORS(CorsFilter.java:290)
	at org.apache.catalina.filters.CorsFilter.doFilter(CorsFilter.java:161)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:192)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:165)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:198)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:108)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:472)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:79)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:349)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:784)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:802)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1410)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by maxifischer** [comment]
What algorithm and data source did you use? It seems to work for most of the algorithms.

**Comment by sekruse** [comment]
It seems that the code that creates histograms admits `null`s: https://github.com/HPI-Information-Systems/Metanome/blob/master/backend/src/main/java/de/metanome/backend/result_postprocessing/helper/ColumnInformation.java#L148
while the crashing part don't: https://github.com/HPI-Information-Systems/Metanome/blob/master/backend/src/main/java/de/metanome/backend/result_postprocessing/helper/ColumnInformation.java#L307
Probably, an `if (entry.getKey() != null)` already does the trick.

**Comment by maxifischer** [comment]
Fixed in 8bc1303f1937d4919e800ec21013a5d6472e9a19
